lib/lightning/workflows.ex

```diff
@@ -405,4 +405,13 @@ defmodule Lightning.Workflows do
   def jobs_ordered_subquery do
     from(j in Job, order_by: [asc: j.inserted_at])
   end
+
+  def latest?(%Workflow{lock_version: version}) do
+    query =
+      from s in Snapshot,
+        where: s.lock_version > ^version,
+        select: count(s.id)
```

Comment:
This could be improved to just consider 1 record, the existance of _anything_ higher than the current version is good enough.

Perhaps:

```elixir
from(s in Snapshot,
  where: s.lock_version > ^version,
  order_by: [desc: s.lock_version],
  limit: 1
  select: true) |> Repo.exists?
```

Reply:
Fixed